components/builds/BuildFilters.tsx

```diff
@@ -19,16 +18,98 @@ import {
   UseBuildsByWeaponState,
 } from "hooks/builds";
 import { useMyTheme } from "hooks/common";
-import { Fragment, useState } from "react";
+import React, { Fragment, useState } from "react";
 import { FiTrash } from "react-icons/fi";
 import { abilities, isMainAbility } from "utils/lists/abilities";
+import { components } from "react-select";
+import ModeImage from "components/common/ModeImage";
+import MySelect from "components/common/MySelect";
 
 interface Props {
   filters: UseBuildsByWeaponState["filters"];
   dispatch: UseBuildsByWeaponDispatch;
 }
 
+const modeOptions = [
+  {
+    label: "All modes",
+    value: "ALL",
+  },
+  {
+    label: "Splat Zones",
+    value: "SZ",
+  },
+  {
+    label: "Tower Control",
+    value: "TC",
+  },
+  {
+    label: "Rainmaker",
+    value: "RM",
+  },
+  {
+    label: "Clam Blitz",
+    value: "CB",
+  },
+  {
+    label: "Turf War",
+    value: "TW",
+  },
+] as const;
+
+const abilitiesOptions = abilities.map((item) => {
+  return { label: item.name, value: item.code };
+});
+
+const ModeOption = (props: any) => {
+  return (
+    <components.Option {...props}>
+      <Flex alignItems="center">
+        <Box mr="0.5em">
+          {props.value !== "ALL" ? (
+            <ModeImage size={24} mode={props.value} />
+          ) : (
+            <></>
+          )}
+        </Box>
+        {props.label}
+      </Flex>
+    </components.Option>
+  );
+};
+
+const AbilityOption = (props: any) => {
+  return (
+    <components.Option {...props}>
+      <Flex alignItems="center">
+        <Box mr="0.5em">
+          <AbilityIcon ability={props.value} size={"SUBTINY"} />
```

Comment:
Useless brackets (will setup a linter soon that will catch these type of issues but bear with me for now 💦 )